sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1929,20 +1933,32 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
                             Context->Devices[0]->ZeDeviceProperties.flags &
                                 ZE_DEVICE_PROPERTY_FLAG_INTEGRATED;
 
+  bool AllocHostPtr = Flags & PI_MEM_FLAGS_HOST_PTR_ALLOC;
+
   if (DeviceIsIntegrated) {
     ze_host_mem_alloc_desc_t ZeDesc = {};
     ZeDesc.flags = 0;
 
     ZE_CALL(zeMemAllocHost(Context->ZeContext, &ZeDesc, Size, 1, &Ptr));
 
-  } else {
-    ze_device_mem_alloc_desc_t ZeDesc = {};
-    ZeDesc.flags = 0;
-    ZeDesc.ordinal = 0;
+  } else if (AllocHostPtr) {
+    // Currently L0 does not support allocation of pinned
+    // host memory. So for PI_MEM_FLAGS_HOST_PTR_ALLOC flag, it allocates
+    // from host accessible memory.
+    ze_host_mem_alloc_desc_t ZeHostMemDesc = {};
+    ZeHostMemDesc.flags = 0;
 
-    ZE_CALL(
-        zeMemAllocDevice(Context->ZeContext, &ZeDesc, Size, 1, ZeDevice, &Ptr));
+    ZE_CALL(zeMemAllocShared(Context->ZeContext, &ZeDeviceMemDesc,
```

Comment:
If the `DeviceIsIntegrated` then the `zeMemAllocHost` is sufficient, and I believe is faster. Otherwise, as requested by user, the allocation should be accessible from the host, so `zeMemAllocSahred` seems good then. I am not sure what will be the interaction of such allocated buffer with the map/unmap operations, whose implementations are highly coupled with this create one. Please add a functional test.

Reply:
there is a test for pinned memory https://github.com/intel/llvm/pull/2080/files?file-filters%5B%5D=.cpp#diff-c0fd7ed69152328e50c4747dc8870b42d5e085b05a3f20cba5c42f79a4002e14